client/ayon_maya/api/lib.py

```diff
@@ -4544,3 +4557,85 @@ def get_scene_units_settings(project_settings=None)-> tuple[str, str]:
     linear_unit = scene_units.get("linear_units", "cm")
     angular_unit = scene_units.get("angular_units", "deg")
     return linear_unit, angular_unit
+
+
+def connect_texture_reference_objects(
+        texture_connections, nodes_by_id, tx_ref_nodes_by_id):
+    """Connect texture reference object nodes to the target object
+    nodes if there is one.
+
+    Args:
+        texture_connections (list): list of texture reference
+        objects connection data
+        nodes_by_id (dict): The dict with node ids.
+        tx_ref_nodes_by_id (dict): The dict with texture reference node ids.
+    """
+    # Compare loaded connections to scene.
+    for texture_connection in texture_connections:
+        source_node = next(
+            iter(tx_ref_nodes_by_id.get(
+            texture_connection["sourceID"], [])
+            ), None
+        )
+
+        target_node = next(
+            iter(nodes_by_id.get(texture_connection["destinationID"], [])
+            ), None
+        )
+
+
+        if not source_node or not target_node:
+            self.log.debug(
+                "Could not find nodes for reference input:\n" +
+                json.dumps(texture_connection, indent=4, sort_keys=True)
+            )
+            continue
+        source_attr, target_attr = texture_connection["connections"]
+
+        if not cmds.attributeQuery(
+            source_attr, node=source_node, exists=True
+        ):
+            self.log.debug(
+                "Could not find attribute {} on node {} for "
+                "reference input:\n{}".format(
+                    source_attr,
+                    source_node,
+                    json.dumps(
+                        texture_connection, indent=4, sort_keys=True
+                    )
+                )
+            )
+            continue
+
+        if not cmds.attributeQuery(
+            target_attr, node=target_node, exists=True
+        ):
+            self.log.debug(
+                "Could not find attribute {} on node {} for "
+                "reference input:\n{}".format(
+                    target_attr,
+                    target_node,
+                    json.dumps(
+                        texture_connection, indent=4, sort_keys=True
+                    )
+                )
+            )
```

Comment:
Shouldn't this `continue` or skip it then? :) Or should this continue to just error?